subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/modulecache/ModuleComponentResolveMetadataSerializer.java

```diff
@@ -121,15 +120,15 @@ public void write(Encoder encoder, ModuleComponentResolveMetadata value) throws
         }
     }
 
-    private void writeOwners(Encoder encoder, ImmutableList<? extends ComponentIdentifier> platformOwners) throws IOException {
+    private void writeOwners(Encoder encoder, ImmutableList<? extends VirtualComponentIdentifier> platformOwners) throws IOException {
         encoder.writeSmallInt(platformOwners.size());
         for (ComponentIdentifier platformOwner : platformOwners) {
-            writeComponentIdentifier(encoder, (ModuleComponentIdentifier)platformOwner);
+            writeComponentIdentifier(encoder, (ModuleComponentIdentifier) platformOwner);
         }
     }
 
     private void writeComponentIdentifier(Encoder encoder, ModuleComponentIdentifier platformOwner) throws IOException {
-        encoder.writeBoolean(platformOwner instanceof VirtualComponentIdentifier);
+        encoder.writeBoolean(true);
```

Comment:
I did not completely remove this as I don't want to change the cache version for 6.1 only because of this.